adapters/algorix/algorix.go

```diff
@@ -0,0 +1,171 @@
+package algorix
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type Adapter struct {
+	EndpointTemplate template.Template
+}
+
+// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	endpoint, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	bidder := &Adapter{
+		EndpointTemplate: *endpoint,
+	}
+	return bidder, nil
+}
+
+// MakeRequests Make Requests
+func (adapter *Adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+	var errs []error
+
+	adapterRequest, err := adapter.makeRequest(request)
+	if err == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	} else {
+		errs = append(errs, err)
+	}
+	return adapterRequests, errs
+}
+
+func (adapter *Adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	if len(request.Imp) == 0 {
```

Comment:
We could remove this statement because prebid server core does this check upstream. Therefore, no request with an empty `Imp` array will get to this point.